hmiddleware/basicauth/checker.go

```diff
@@ -50,7 +50,14 @@ func parseCredential(credential string) (Credential, error) {
 // Checker stores a set of valid credentials to be used when authenticating
 // HTTP requests using basic auth.
 type Checker struct {
-	credentials []Credential
+	credentials   []Credential
+	roleValidator RoleValidator
```

Comment:
Is it possible we could want more than one role validator?

Reply:
Mm I can't see a reason we would. The Credentials field is a list because they are pairs of username/passwords, where each client gets its own pair.